site/docs/plugins/commands.md

```diff
@@ -5,10 +5,628 @@ next: false
+
+Now, Command Scopes are a cool feature, but using it by hand can get really messy, since it's hard to keep track of all the scopes and what commands they present.
+Plus, by using Command Scopes on their own, you have to do manual filtering inside each command to ensure they'll only run for the correct scopes.
+Syncing those two things up can be a nightmare, and that's why this plugin exists.
+Check how it's done.
+
+The `Command` class returned by the `command` method exposes a method called `addToScope`.
+This method takes in a [BotCommandScope](/ref/types/botcommandscope) together with one or more handlers, and registers those handlers to be ran at that specific scope.
+
+You don't even need to worry about calling `filter`, the `addToScope` method will guarantee that your handler only gets called if the context is right.
+
+Here's an example of a scoped command:
+
+```ts
+const myCommands = new CommandGroup();
+
+myCommands
+  .command("start", "Initializes bot configuration")
+  .addToScope(
+    { type: "all_private_chats" },
+    (ctx) => ctx.reply(`Hello, ${ctx.chat.first_name}!`),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply(`Hello, members of ${ctx.chat.title}!`),
+  );
+```
+
+The `start` command can now be called from both private and group chats, and it will give a different response depending on where it gets called from.
+Now if you call `myCommands.setCommands`, the `start` command will be registered to both private and group chats.
+
+Here's an example of a command that's only accessible to group admins.
+
+```js
+adminCommands
+  .command("secret", "Admin only")
+  .addToScope(
+    { type: "all_chat_administrators" },
+    (ctx) => ctx.reply("Free cake!"),
+  );
+```
+
+And here is an example of a command that's only accessible in groups
+
+```js
+myCommands
+  .command("fun", "Laugh")
+  .addToScope(
+    { type: "all_group_chats" },
+    (ctx) => ctx.reply("Haha"),
+  );
+```
+
+Notice that when you call the `command` method, it opens up a new command.
+If you give it a handler, that handler will apply to the `default` scope of that command.
+Calling `addToScope` on that command will then add a new handler, which will be filtered to that scope.
+Take a look at this example.
+
+```ts
+myCommands
+  .command(
+    "default",
+    "Default command",
+    // This will be called when not on a group chat, or when the user is not an admin
+    (ctx) => ctx.reply("Hello from default scope"),
+  )
+  .addToScope(
+    { type: "all_group_chats" },
+    // This will only be called for non-admin users in a group
+    (ctx) => ctx.reply("Hello, group chat!"),
+  )
+  .addToScope(
+    { type: "all_chat_administrators" },
+    // This will be called for group admins, when inside that group
+    (ctx) => ctx.reply("Hello, admin!"),
+  );
+```
```

Comment:
Somehow, I’m always getting the default response, whether I’m an admin or not.

Steps:
1. Sent `/default` to the bot in a private chat and got back: `Hello from default scope`.
2. Sent `/default` in a group where I’m the creator (admin), and got the same response: `Hello from default scope`.
3. A non-admin member sent `/default` in the group and also got: `Hello from default scope` too.

Here is my code:
```ts
import { Bot } from "https://deno.land/x/grammy@v1.31.0/mod.ts";
import { CommandGroup } from "https://deno.land/x/grammy_commands@v0.10.0/mod.ts";

const botToken = Deno.env.get("BOT_TOKEN");
if (!botToken) throw new Error("Missing bot token!");

const bot = new Bot(botToken);

const myCommands = new CommandGroup();

myCommands
  .command(
    "default",
    "Default command",
    // This will be called when not in a group chat
    (ctx) => ctx.reply("Hello from default scope"),
  )
  .addToScope(
    { type: "all_group_chats" },
    // This will only be called for non-admin users in a group
    (ctx) => ctx.reply("Hello, group chat!"),
  )
  .addToScope(
    { type: "all_chat_administrators" },
    // This will be called for group admins, when inside that group
    (ctx) => ctx.reply("Hello, admin!"),
  );

bot.use(myCommands);
await myCommands.setCommands(bot);

bot.start({
  onStart() {
    console.info("Ready.");
  },
});
```

Reply:
I know what the problem is. We need to add the default scope as the last handler internally. We can move on with the docs PQ and I'll open an issue on the plug-in repo to fix this before we release.

Reply:
This snippet also needs to change the order in which the scopes are added. "all_chat_administrators" first and then "all_group_chats"